chapter/background.md

```diff
@@ -0,0 +1,607 @@
+4. Find references
+   List all usages of a defined variable.
+5. Workspace/Document symbols
+   List all variables in a workspace or document.
+6. Diagnostics
+   Analyze source code, i.e., parse and type check and notify the LSP Client if errors arise.
+
+#### Code Completion
+
+This feature allows users to request a suggested identifier of variables or methods, concrete values or larger templates of generated code to be inserted at the position of the cursor.
+The completion can be invoked manually or upon entering language defined trigger characters, such as `.`, `::` or `->`.
+The completion request contains the current cursor position, allowing the language server to resolve contextual information based on an internal representation of the document.
+In the example ([@fig:lsp-capability-complete]) the completion feature suggests related identifiers for the incomplete function call "`pr`*`int`*".
+
+![](examples/complete.png){#fig:lsp-capability-hover caption="Completion options resolved by the Python language server in Visual Studio Code"}
+
+
+#### Hover
+
+Hover requests are issued by editors when the user rests their mouse cursor on text in an opened document or issues a designated command in editors without mouse support.
+If the language server has indexed any information corresponding to the position, it can generate a report using plain text and code elements, which are then rendered by the editor.
+Language servers typically use this to communicate type-signatures or documentation.
+An example can be seen in [@fig:lsp-capability-hover].
+
+![](examples/hover.png){#fig:lsp-capability-hover caption="Hover information displayed by the Python language server in Visual Studio Code"}
+
+
+#### Jump to Definition
+
+The LSP allows users to navigate their code by the means of symbols by finding the definition of a requested symbol.
+Symbols can be for instance variable names or function calls.
+As seen in [@fig:lsp-capability-definition], editors can use the available information to enrich hover overlays with the hovered element's definition.
+
+![](examples/definition.png){#fig:lsp-capability-hover caption="Hover information enriched with the definition. Python language server in Visual Studio Code"}
+
+
+#### Find References
+
+Finding references is the inverse operation to the previously discussed *Jump to Definition* ([cf. @sec:jump-to-definition]).
+For a given symbol definition, for example variable, function, function argument or record field the LSP provides all usages of the symbol allowing users to inspect or jump to the referencing code position.
+
+![](examples/references.png){#fig:lsp-capability-hover caption="Listing of all references to the method "hello". Python language server in Visual Studio Code"}
+
+#### Workspace/Document symbols
+
+The symbols capability allows language servers to expose a list if symbols declared in the open document or workspace.
```

Comment:
clearly define "document" and "workspace" in this context first.

Reply:
```suggestion
The symbols capability is defined as both a "Language Feature" and "Workspace Feature" which mainly differ in the scope they represent.
The `textDocument/documentSymbol` command lists symbols solely found in the currently opened file, while the `workspace/symbol` takes into account all files in the opened set of folders.
```